code/modules/projectiles/guns/energy/experimental.dm

```diff
@@ -0,0 +1,21 @@
+/obj/item/weapon/gun/energy/ricochet
+	name = "ricochet rifle"
+	desc = "They say that ducks made this weapon. Yes, the quacking type."
+	icon = 'icons/obj/gun_experimental.dmi'
+	icon_state = "ricochet"
+	item_state = null
+	projectile_type = "/obj/item/projectile/ricochet"
+	charge_cost = 100
+	cell_type = "/obj/item/weapon/cell"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/experimentals.dmi', "right_hand" = 'icons/mob/in-hand/right/experimentals.dmi')
+
+/obj/item/weapon/gun/energy/bison
+	name = "righteous bison"
```

Comment:
A few, but they're mostly related to people not stripping names properly before trying to put them into UIs. It doesn't have a noticeable impact otherwise.